File: VB/WindowsApplication1/CustomControls/CustomListBoxControl/CustomListBoxControl.vb
Imports System.ComponentModel
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports System

Namespace Example
	<ToolboxItem(True)>
	Public Class CustomListBoxControl
		Inherits ListBoxControl

		Protected Overrides Function CreateViewInfo() As BaseStyleControlViewInfo
			Return New CustomListBoxViewInfo(Me)
		End Function

		#Region "ShiftItem"
		Public Event ShiftItem As EventHandler(Of ItemInfoEventArgs)
			Protected Friend Sub RaiseShiftItem(ByVal ea As ItemInfoEventArgs)
            RaiseEvent ShiftItem(Me, ea)
        End Sub
		#End Region
	End Class
End Namespace

File: CS/WindowsApplication1/CustomControls/CustomListBoxControl/CustomListBoxControl.cs
using System.ComponentModel;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.ViewInfo;
using System;

namespace Example
{
    [ToolboxItem(true)]
    public class CustomListBoxControl : ListBoxControl
    {
        protected override BaseStyleControlViewInfo CreateViewInfo() { return new CustomListBoxViewInfo(this); }
 
        #region ShiftItem
        public event EventHandler<ItemInfoEventArgs> ShiftItem;
        protected internal void RaiseShiftItem(ItemInfoEventArgs ea)
        {
            EventHandler<ItemInfoEventArgs> handler = ShiftItem;
            if (handler != null)
                handler(this, ea);
        }
        #endregion
    }
}
